app/models/hyku_addons/csv_entry.rb

```diff
@@ -33,6 +34,22 @@ def build_metadata
       parsed_metadata
     end
 
+    def add_file_subfields
+      file_metadata = []
+      raw_metadata.select { |k, _v| k =~ /^file_((?<subfield>.+)_)?(?<index>\d+)$/ }.each do |k, v|
+        match = k.match(/^file_((?<subfield>.+)_)?(?<index>\d+)$/)
+        file_metadata[match[:index].to_i - 1] ||= {}
```

Comment:
I think this will add some clarity to this sentence and the next ones while also avoiding some code repetition.
```suggestion
        file_index = match[:index].to_i - 1
        file_metadata[file_index] ||= {}
```